src/quickpicks/schemas.ts

```diff
@@ -73,3 +99,254 @@ export async function schemaTypeQuickPick(): Promise<SchemaType | undefined> {
 
   return chosenType as SchemaType;
 }
+
+/** Quickpick over the versions of a schema based on its subject. */
+export async function schemaVersionQuickPick(
+  schemaRegistry: SchemaRegistry,
+  subject: string,
+): Promise<Schema | undefined> {
+  const loader = ResourceLoader.getInstance(schemaRegistry.connectionId);
+  const schemaVersions: Schema[] = await loader.getSchemasForEnvironmentId(
+    schemaRegistry.environmentId,
+  );
+  const schemasMatchingSubject: Schema[] = schemaVersions.filter(
+    (schema) => schema.subject === subject,
+  );
+  if (schemasMatchingSubject.length === 1) {
+    // skip the quickpick if there's only one version
+    return schemasMatchingSubject[0];
+  }
+
+  const versionItems: vscode.QuickPickItem[] = schemasMatchingSubject.map((schema) => ({
+    label: `v${schema.version.toString()}`,
+    description: schema.isHighestVersion ? `${schema.id} (latest)` : schema.id,
+  }));
+  const chosenVersionItem: vscode.QuickPickItem | undefined = await vscode.window.showQuickPick(
+    // sort by version in descending order (latest at the top)
+    versionItems.sort((a, b) => b.label.localeCompare(a.label)),
+    {
+      title: `Schema versions for ${subject}`,
+      placeHolder: "Select a schema version",
+      ignoreFocusOut: true,
+    },
+  );
+
+  if (!chosenVersionItem) {
+    return;
+  }
+  return schemaVersions.find(
+    (schema) =>
+      schema.id === chosenVersionItem.description?.split(" ")[0] &&
+      schema.version === parseInt(chosenVersionItem.label.replace("v", ""), 10),
+  );
+}
+
+export type SchemaKindSelection = {
+  keySchema: boolean;
+  valueSchema: boolean;
+  deferToDocument: boolean;
+};
+
+/**
+ * Quickpick to (multi-)select which schema kind (key and/or value).
+ *
+ * If the provided `topic` is already associated with a schema based on the `TopicNameStrategy`,
+ * pre-select that kind.
+ *
+ * Deselecting any pre-selected kind(s) will show a confirmation warning modal to confirm the user
+ * wants to produce without schema(s).
+ */
+export async function schemaKindMultiSelect(
```

Comment:
This function is kinda long to read. Consider breaking up or pull the element lists into a different form